packages/angular/src/jsonforms-root.component.ts

```diff
@@ -0,0 +1,105 @@
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import {
+    Component,
+    EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges
+} from '@angular/core';
+import { generateDefaultUISchema, JsonFormsRendererRegistryEntry, JsonSchema, UISchemaElement, UISchemaTester } from '@jsonforms/core';
+import { JsonFormsAngularService } from './jsonforms.service';
+@Component({
+    selector: 'jsonforms',
+    template: '<jsonforms-outlet></jsonforms-outlet>',
+    providers: [JsonFormsAngularService]
+})
+export class JsonForms implements OnChanges, OnInit {
+
+    @Input() uischema: UISchemaElement;
+    @Input() schema: JsonSchema;
+    @Input() data: any;
+    @Input() renderers: JsonFormsRendererRegistryEntry[];
+    @Input() uischemas: { tester: UISchemaTester; uischema: UISchemaElement; }[];
+    @Output() dataChange = new EventEmitter<any>();
+    @Input() locale: string;
+    @Input() readonly: boolean;
+
+    constructor(private jsonformsService: JsonFormsAngularService) {
+    }
+
+    ngOnInit(): void {
+        this.jsonformsService.init({
+            core: {
+                data: this.data,
+                uischema: this.uischema ?? generateDefaultUISchema(this.schema),
+                schema: this.schema
+            },
+            uischemas: this.uischemas,
+            i18n: { locale: this.locale, localizedSchemas: undefined, localizedUISchemas: undefined },
+            renderers: this.renderers
+        });
+        this.jsonformsService.$state.subscribe(state => {
+            const data = state?.jsonforms?.core?.data;
+            if (data) {
+                this.dataChange.emit(data);
+            }
+        });
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const newData = changes.data;
+        if (newData && !newData.isFirstChange()) {
+            this.jsonformsService.setData(newData.currentValue);
+        }
+
+        const newSchema = changes.schema;
+        if (newSchema && !newSchema.isFirstChange()) {
+            this.jsonformsService.setSchema(newSchema.currentValue);
+        }
+
+        const newUiSchema = changes.uischema;
+        if (newUiSchema && !newUiSchema.isFirstChange()) {
+            this.jsonformsService.setUiSchema(newUiSchema.currentValue);
+        }
```

Comment:
[In React](https://github.com/eclipsesource/jsonforms/blob/a51e3736ddc33a595887f29263dcb9242608a5d8/packages/react/src/JsonFormsContext.tsx#L124-L128) we use the `updateCore` action here as we didn't want to update (and trigger validation, emit events etc.) three separate times when `data`, `schema` and `uischema` change at the same time. Can we implement this same behavior here too?

Reply:
probably yes, I check